File: 1508-range-sum-of-sorted-subarray-sums/1508-range-sum-of-sorted-subarray-sums.py
class Solution:
    def rangeSum(self, nums, n, left, right):
        mod = 10**9 + 7
        arr = []
        for i in range(len(nums)):
            sum_ = 0
            for j in range(i, len(nums)):
                arr.append(nums[j] + sum_)
                sum_ += nums[j]
                
        arr.sort()
        
        total = sum(arr[left-1:right])
        
        return total%mod
        
                
            
        

File: 1508-range-sum-of-sorted-subarray-sums/1508-range-sum-of-sorted-subarray-sums.go
func rangeSum(nums []int, n int, left int, right int) int {
    var subarray []int
    
    const MOD = 1000000007

    for i := 0; i < len(nums); i++ {
        sum := 0
        for j := i; j < len(nums); j++ {
            sum += nums[j]
            subarray = append(subarray, sum)
        }
    }

    sort.Ints(subarray)
    total := 0

    for i := left - 1; i < right; i++ {
        total += subarray[i]
    }

    return total%MOD
}